packages/paymaster/Readme.md

```diff
@@ -0,0 +1,183 @@
+import { IPaymaster, BiconomyPaymaster } from '@biconomy/paymaster'
+
+  const paymaster = new BiconomyPaymaster({
+    paymasterUrl: '' // you can get this value from biconomy dashboard. https://dashboard.biconomy.io
+  })
+```
+
+**paymasterUrl** you can get this value from biconomy dashboard.
+
+Following are the methods that can be called on paymaster instance 
+
+```typescript
+export interface IHybridPaymaster<T> extends IPaymaster {
+  getPaymasterAndData(
+    userOp: Partial<UserOperation>,
+    paymasterServiceData?: T
+  ): Promise<PaymasterAndDataResponse>
+  buildTokenApprovalTransaction(
+    tokenPaymasterRequest: BiconomyTokenPaymasterRequest,
+    provider: Provider
+  ): Promise<Transaction>
+  getPaymasterFeeQuotesOrData(
+    userOp: Partial<UserOperation>,
+    paymasterServiceData: FeeQuotesOrDataDto
+  ): Promise<FeeQuotesOrDataResponse>
+}
+```
+
+**getPaymasterAndData**
+
+This function accepts a **`Partial<UserOperation>`** object that includes all properties of **`userOp`** except for the **`signature`** field. It returns **`paymasterAndData`** as part of the **`PaymasterAndDataResponse`**. The **`paymasterAndData`** string is signed by the paymaster's verifier signer, which will eventually result in the payment of your transaction fee by the paymaster.
+
+**buildTokenApprovalTransaction**
+
+This function is specifically used for token paymaster sponsorship. The primary purpose of this function is to create an approve transaction that gets batched with the rest of your transactions. This way, you will be paying the paymaster in ERC20 tokens, which will result in the paymaster paying on your behalf in native tokens.
+
+Note: you don’t need to call this function. it will automatically get **buildTokenPaymasterUserOp** using account package
+
+**getPaymasterFeeQuotesOrData**
+
+This function is used to fetch quote information or data base on provider userOperation and paymasterServiceData. 
+
+```typescript
+const feeQuotesResponse =
+        await biconomyPaymaster.getPaymasterFeeQuotesOrData(partialUserOp, {
+          mode: PaymasterMode.ERC20,
+          calculateGasLimits: true,
+          tokenList: [],
+          preferredToken: "" // you can also supply preferred token value
+        });
+```
+
+If you supply specific address in tokenList array. System will return fee quotes for those specific token. Alternatively, you can supply preferredToken address you wants to pay in. 
+
+Note: This function can return **paymasterAndData** as well in case all of the policies checks get pass.
+
+Here is Token paymaster sponsorship example in typescript
+
+```typescript
+
+import { BiconomySmartAccount, BiconomySmartAccountConfig, DEFAULT_ENTRYPOINT_ADDRESS } from "@biconomy/account"
+
+import {
+    IHybridPaymaster,
+    BiconomyPaymaster,
+    PaymasterFeeQuote,
+    PaymasterMode,
+    SponsorUserOperationDto,
+} from '@biconomy/paymaster'
+
+const bundler: IBundler = new Bundler({
```

Comment:
this readme should only showcase how to create paymaster instance and different methods available. and minimal info on how to use it with userop. actually just say that once you get paymasterAndData you can update your userop. 
All specific about attributes passed in case of hybrid paymaster can be added separately here.